sktime/performance_metrics/forecasting/_functions.py

```diff
@@ -135,3 +139,109 @@ def mape_loss(y_test, y_pred):
     eps = np.finfo(np.float64).eps
 
     return np.mean(np.abs(y_test - y_pred) / np.maximum(np.abs(y_test), eps))
+
+
+def evaluate(
+    forecaster, cv, y, X=None, strategy="refit", scoring=smape_loss, return_data=False
+):
+    """Evaluate forecaster using cross-validation
+
+    Parameters
+    ----------
+    forecaster : sktime.forecaster
+        Any forecaster
+    cv : sktime.SlidingWindowSplitter or sktime.ExpandingWindowSplitter
+        Splitter of how to split the data into test data and train data
+    y : pd.Series
+        Target time series to which to fit the forecaster.
+    X : pd.DataFrame, optional (default=None)
+        Exogenous variables (ignored)
+    strategy : str, optional
+        Must be "refit" or "update", by default "refit". The strategy defines
+        whether forecaster is only fitted on the first train window data and
+        then updated or always refitted.
+    scoring : function, optional
+        A score function that takes y_pred and y_test as arguments,
+        by default smape_loss
+    return_data : bool, optional
+        Returns three additional columns in the DataFrame, by default False.
+        The cells of the columns contain each a pd.Series for y_train,
+        y_pred, y_test.
+
+    Returns
+    -------
+    pd.DataFrame
```

Comment:
I'm somewhat torn about this return signature.

Should there not be an easier way to get average performance across folds?
Here, you would have to write sth like `evaluate(foo)['test_smape_loss'].mean()` instead of `evaluate(foo)` or `evaluate(foo, aggregate=sth)`, which seems cumbersome.

What do you think, @mloning?

Reply:
We could add aggregation to the evaluate function but we may still want to return the raw data so we would add another option. I think for now it's preferably to leave it to the user to aggregate over rows. I think this still works well:
```python
results = evaluate(...)
results["test_smape_loss"].mean()
```

Reply:
hm, would you say that computing aggregate scores is a concern that better fits with benchmarking than with `evaluate`?

Reply:
@fkiraly yes, at least for now